### PR TITLE
[rb] Add `base64` gem as a runtime dependency

### LINKED ISSUE #13447: [🐛 Bug]: [rb] Either declare or remove dependency on `base64`
### What happened?

Ruby 3.3 shows a warning if the base64 gem is required without it being declared in the gemspec, Ruby 3.4 will throw an error.
 See https://github.com/ruby/ruby/blob/master/doc/NEWS/NEWS-3.3.0.md#stdlib-updates

Many gems have opted to inline the base64 implementation, however this gem makes use of the urlsafe variants which are a bit more complex and not just a wrapper over `pack`/`unpack`. You would need to inline the entirety of https://github.com/ruby/base64/blob/9669a7d3b0e3b9a739969404daf58f912c58c6b3/lib/base64.rb, similar to https://github.com/newrelic/newrelic-ruby-agent/pull/2378

### How can we reproduce the issue?

```shell
Use `selenium-webdriver` on Ruby 3.3
```

### Relevant log output

```shell
base64 is not part of the default gems since Ruby 3.4.0. Add it to your Gemfile.
```

### Operating System

Irrelevant

### Selenium version

Ruby master

### What are the browser(s) and version(s) where you see this issue?

Irrelevant

### What are the browser driver(s) and version(s) where you see this issue?

Irrelevant

### Are you using Selenium Grid?

_No response_

### PR DESCRIPTION
**Pull request by Earlopain**
### Description

Closes #13447

Ruby 3.3 shows a warning if requiring base64 without specifying it in the gemspec
Ruby 3.4 will throw a warning.
See https://github.com/ruby/ruby/blob/master/doc/NEWS/NEWS-3.3.0.md#stdlib-updates

There is no version constraint on base64. This follows what other projects have done:
* https://github.com/rails/rails/pull/48907
* ~~https://github.com/rubocop/rubocop/pull/12094~~ Nevermind this one, I misremembered
* https://github.com/newrelic/newrelic-ruby-agent/pull/2238

There may be some issue with this in combination with older bundler versions. But since this gem requires Ruby >= 3.0 I don't think this is a concern.
* https://github.com/rubocop/rubocop/issues/12310
* https://github.com/newrelic/newrelic-ruby-agent/issues/2312

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rb/Gemfile.lock [hunk 1]
```diff
@@ -4,6 +4,7 @@ PATH
     selenium-devtools (0.120.0)
       selenium-webdriver (~> 4.2)
     selenium-webdriver (4.17.0.nightly)
+      base64
```
Comment (titusfortner) [inline rb/Gemfile.lock]:
```suggestion
      base64 (~> 0.2)
```

**Comment by titusfortner** [comment]
Excellent. Can you add the version and re-run bundler for the lockfile? Thanks.

**Comment by Earlopain** [comment]
Yeah, sure. Done

**Comment by titusfortner** [comment]
@Earlopain Thanks for your help!

**Pull request merged**